sources/cassandra/validations/count.go

```diff
@@ -0,0 +1,436 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"math/big"
+	"os"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"cloud.google.com/go/spanner"
+	"github.com/gocql/gocql"
+	"google.golang.org/api/iterator"
+)
+
+// This script connects to both Cassandra and Spanner databases to count the number of rows in specified tables.
+// It allows for parallel processing of the counting operation using multiple workers.
+//
+// Sample usage:
+// go run count.go -host localhost -port 9042 -keyspace my_keyspace -spanner-uri projects/my_project/instances/my_instance/databases/my_database
+func main() {
+	// Define command line arguments
+	host := flag.String("host", "localhost", "Cassandra host")
+	port := flag.Int("port", 9042, "Cassandra port")
+	keyspace := flag.String("keyspace", "", "Keyspace name")
+	username := flag.String("username", "", "Cassandra username")
+	password := flag.String("password", "", "Cassandra password")
+	table := flag.String("table", "", "Table name (empty for all tables)")
+	workers := flag.Int("workers", 8, "Number of parallel workers")
+	spannerURI := flag.String("spanner-uri", "", "Spanner database URI (projects/PROJECT_ID/instances/INSTANCE_ID/databases/DATABASE_ID)")
+	flag.Parse()
+
+	if *keyspace == "" {
+		fmt.Println("keyspace name is required")
+		os.Exit(1)
+	}
+
+	if *spannerURI == "" {
+		fmt.Println("spanner-uri is required")
+		os.Exit(1)
+	}
+
+	// Connect to Cassandra
+	cassSession, err := connectToCassandra(*host, *port, *keyspace, *username, *password)
+	if err != nil {
+		fmt.Printf("Failed to connect to Cassandra: %v\n", err)
+		os.Exit(1)
+	}
+	defer cassSession.Close()
+
+	// Connect to Spanner
+	ctx := context.Background()
+	spannerClient, err := connectToSpanner(ctx, *spannerURI)
+	if err != nil {
+		fmt.Printf("Failed to connect to Spanner: %v\n", err)
+		os.Exit(1)
+	}
+	defer spannerClient.Close()
+
+	tables := []string{}
+	if *table != "" {
+		tables = append(tables, *table)
```

Comment:
[nit feel free to close with a todo] 
Please consider taking an inner join of tables on spanner and cassandra table names and validate tables that have been created on both.

Reply:
Done